frontend/model/state.js

```diff
@@ -329,6 +335,235 @@ const getters = {
+  },
+  groupIncomeAdjustedDistributionForMonth (state, getters) {
+    return monthstamp => groupIncomeDistribution({ state, getters, monthstamp, adjusted: true })
+  },
+  ourPayments (state, getters) {
+    const currency = currencies[getters.groupSettings.mincomeCurrency]
+    const ourUsername = getters.ourUsername
+    const cMonthstamp = currentMonthstamp()
+    const pDate = dateFromMonthstamp(cMonthstamp)
+    const dueIn = lastDayOfMonth(pDate)
+    const monthlyPayments = getters.groupMonthlyPayments
+    const allPayments = getters.currentGroupState.payments
+
+    const sent = (() => {
+      const payments = []
+      // QUESTION: Is this only from this month? What about the previous payments from months ago?
+      //           Same question for received.
+      for (const monthstamp of Object.keys(monthlyPayments).sort()) {
+        const { paymentsFrom } = monthlyPayments[monthstamp]
+
+        if (!paymentsFrom) { continue }
+
+        for (const toUser in paymentsFrom[getters.ourUsername]) {
+          for (const paymentHash of paymentsFrom[getters.ourUsername][toUser]) {
+            const { data, meta } = allPayments[paymentHash]
+
+            payments.push({ hash: paymentHash, data, meta })
+          }
+        }
+      }
+      return payments
+    })()
+    const received = (() => {
+      const thisMonthPayments = monthlyPayments[cMonthstamp]
+      const paymentsFrom = thisMonthPayments && thisMonthPayments.paymentsFrom
+      const payments = []
+
+      if (!paymentsFrom) { return [] }
+
+      for (const fromUser in paymentsFrom) {
+        for (const toUser in paymentsFrom[fromUser]) {
+          if (toUser === getters.ourUsername) {
+            for (const paymentHash of paymentsFrom[fromUser][toUser]) {
+              const { data, meta } = allPayments[paymentHash]
+
+              payments.push({ hash: paymentHash, data, meta })
+            }
+          }
+        }
+      }
+
+      return payments
+    })()
+    const todo = (() => {
+      const payments = []
+      const unadjusted = getters.groupIncomeDistribution.filter(p => p.from === ourUsername)
+
+      for (const p of getters.groupIncomeAdjustedDistribution) {
+        if (p.from !== ourUsername) { continue }
+        const existPayment = unadjusted.find(({ to }) => to === p.to) || { amount: 0 }
+        const amount = +currency.displayWithoutCurrency(p.amount)
+        const existingAmount = +currency.displayWithoutCurrency(existPayment.amount)
+
+        if (amount <= 0) { continue }
+
+        const partialAmount = existingAmount - amount
+        const existingPayment = {}
+        if (partialAmount > 0) {
+          // TODO/BUG this only work if the payment is done in 2 parts. if done in >=3 won't work.
+          const sentPartial = sent.find((s) => s.username === p.to && s.amount === partialAmount)
+          if (sentPartial) {
+            existingPayment.hash = sentPartial.hash // Q: Why do we need this hash?
```

Comment:
All payment entries in `payments` have the hash. They need it because otherwise there's no way to know what payment this represents.

Reply:
But we don't use it everywhere in this scenario, that's what I meant...

Reply:
Without passing in the hash it will be impossible to access information about the original payment.